saleor/graphql/giftcard/filters.py

```diff
@@ -113,3 +113,25 @@ def check_currency_in_filter_data(filter_data: dict):
 class GiftCardFilterInput(FilterInputObjectType):
     class Meta:
         filterset_class = GiftCardFilter
+
+
+def filter_events_by_type(events: List[models.GiftCardEvent], type_value: str):
+    filtered_events = []
+    for event in events:
+        if event.type == type_value:
+            filtered_events.append(event)
+    return filtered_events
+
+
+def filter_events_by_order_id(events: List[models.GiftCardEvent], order_id: str):
+    _type, order_pk = from_global_id_or_error(order_id, "Order")
+    filtered_events = []
+    for event in events:
+        if str(event.parameters.get("order_id")) == order_pk:
+            filtered_events.append(event)
+    return filtered_events
+
+
+class GiftCardEventFilterInput(graphene.InputObjectType):
+    type = graphene.Argument(GiftCardEventsEnum)
+    order_id = graphene.Argument(graphene.ID)
```

Comment:
Maybe we could pass a list of ids?

Reply:
There is not need right now for this, but maybe it's better solution to use list here, as it's more flexible

Reply:
IMO we should implement a more flexible solution. In other places, we use a list of ids instead of one id eg. `AttributeFilter.ids`, `ProductFilter.ids`, `ProductFilter.ProductFilter`.